File: src/etch/bindings/csharp/msg/Validator_RuntimeException.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Etch.Msg
{
    /// <summary>
    /// Validator for Runtime Exceptions
    /// </summary>
    public class Validator_RuntimeException : Validator
    {
        /// <summary>
        /// 
        /// </summary>
        /// <returns>an instance of the validator</returns>
        public static Validator_RuntimeException Get()
        {
            if ( validator == null )
                validator = new Validator_RuntimeException();

            return validator;
        }

        private static Validator_RuntimeException validator;

        public Validator_RuntimeException()
        {
            // nothing to do
        }

        public override Validator ElementValidator()
        {
            throw new Exception( "not an array" );
        }

        public override bool Validate( object value )
        {
            return ( value is Exception );
        }

        public override sbyte? CheckValue( object value )
        {
            if ( Validate( value ) )
                return Etch.Transport.Fmt.TypeCode.CUSTOM;
            else
                return null;
        }
    }
}

File: src/etch/bindings/java/msg/Validator_RuntimeException.java
/* $Id$
 *
 * Created by wert on Aug 20, 2007.
 *
 * Copyright (c) 2007 Cisco Systems, Inc. All rights reserved.
 */

package etch.bindings.java.msg;

import etch.bindings.java.transport.fmt.TypeCode;

/**
 * Validator for RuntimeExceptions.
 */
public class Validator_RuntimeException extends Validator
{
	/**
	 * @return an instance of the validator.
	 */
	public static Validator_RuntimeException get()
	{
		if (validator == null)
			validator = new Validator_RuntimeException();
		
		return validator;
	}
	
	private static Validator_RuntimeException validator;

	/**
	 * Constructs the Validator.
	 *
	 */
	public Validator_RuntimeException()
	{
		// nothing to do.
	}

	@Override
    public Validator elementValidator()
    {
	    return null;
    }

	@Override
	final public boolean validate( Object value )
	{
		return value instanceof RuntimeException;
	}

	@Override
	public Byte checkValue( Object value )
	{
		return validate( value ) ? TypeCode.CUSTOM : null;
	}
}
